src/components/PasswordStrengthCheck/PasswordStrengthCheck.tsx

```diff
@@ -0,0 +1,142 @@
+import {
+  DetailedHTMLProps,
+  forwardRef,
+  HTMLAttributes,
+  PropsWithoutRef,
+  useEffect,
+  useState,
+  useCallback,
+} from 'react';
+import { RVColorProp, RVSizeProp, RVVariantProp } from '../../types';
+import { Checkbox } from '../Checkbox';
+import styles from './PasswordStrengthCheck.module.scss';
+
+export interface RVPasswordStrengthCheck
+  extends Omit<
+    PropsWithoutRef<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>>,
+    'color' | 'size'
+  > {
+  variant?: Exclude<RVVariantProp, RVVariantProp.disabled>;
+  color?: RVColorProp;
+  size?: RVSizeProp;
+  password?: string;
+  onSuccessfulValidation?: (password: string, validations: Record<string, boolean>) => void;
+  validations: {
```

Comment:
Please check all of the below items:

we have a settings with the below type definition for password checking:
```tsx
type PasswordPolicySetting = {
  MinLength?: number; //ignore if smaller than 2
  NewCharacters?: number; //ignore if smaller than 2
  UpperLower?: boolean;
  NonAlphabetic?: boolean;
  NonAlphaNumberic?: boolean;
  Number?: boolean;
  SpecialCharacters?: boolean;
}
```